packages/atomic/src/components/common/item-list/item-list-common.spec.ts

```diff
@@ -0,0 +1,301 @@
+
+  describe('#focusOnNextNewResult', () => {
+    it('should call #props.getCurrentNumberOfItems()', () => {
+      const getCurrentNumberOfItems = vi.fn();
+      const itemListCommon = itemListCommonFixture({
+        getCurrentNumberOfItems,
+      });
+
+      itemListCommon.focusOnNextNewResult();
+
+      expect(getCurrentNumberOfItems).toHaveBeenCalled();
+    });
+
+    it('should make #setNewResultRef(e, n) not call #getFirstFocusableDescendant / #props.nextNewItemTarget.setTarget when e has children and n is not the value returned by #props.getCurrentNumberOfItems()', () => {
+      const getFirstFocusableDescendantMock = vi.mocked(
+        getFirstFocusableDescendant
+      );
+      const getCurrentNumberOfItems = vi.fn(() => 5);
+      const nextNewItemTarget = {
+        focus: vi.fn(),
+        focusAfterSearch: vi.fn(),
+        focusOnNextTarget: vi.fn(),
+        setTarget: vi.fn(),
+      };
+      const itemListCommon = itemListCommonFixture({
+        getCurrentNumberOfItems,
+        nextNewItemTarget,
+      });
+
+      itemListCommon.focusOnNextNewResult();
+
+      const element = document.createElement('div');
+      element.appendChild(document.createElement('span'));
+
+      itemListCommon.setNewResultRef(element, 0);
+
+      expect(getFirstFocusableDescendantMock).not.toHaveBeenCalled();
+      expect(nextNewItemTarget.setTarget).not.toHaveBeenCalled();
+    });
+
+    it('should make #setNewResultRef(e, n) call #getFirstFocusableDescendant / #props.nextNewItemTarget.setTarget when e has children and n is equal to the value returned by #props.getCurrentNumberOfItems()', () => {
+      const getFirstFocusableDescendantMock = vi.mocked(
+        getFirstFocusableDescendant
+      );
+      const getCurrentNumberOfItems = vi.fn(() => 5);
+      const nextNewItemTarget = {
+        focus: vi.fn(),
+        focusAfterSearch: vi.fn(),
+        focusOnNextTarget: vi.fn(),
+        setTarget: vi.fn(),
+      };
+      const itemListCommon = itemListCommonFixture({
+        getCurrentNumberOfItems,
+        nextNewItemTarget,
+      });
+
+      itemListCommon.focusOnNextNewResult();
+
+      const element = document.createElement('div');
+      element.appendChild(document.createElement('span'));
+
+      itemListCommon.setNewResultRef(element, 5);
+
+      expect(getFirstFocusableDescendantMock).toHaveBeenCalled();
+      expect(nextNewItemTarget.setTarget).toHaveBeenCalled();
+    });
+
+    it('should call #props.nextNewItemTarget.focusOnNextTarget()', () => {
+      const nextNewItemTarget = {
+        focus: vi.fn(),
+        focusAfterSearch: vi.fn(),
+        focusOnNextTarget: vi.fn(),
+        setTarget: vi.fn(),
+      };
+      const itemListCommon = itemListCommonFixture({
+        nextNewItemTarget,
+      });
+
+      itemListCommon.focusOnNextNewResult();
+
+      expect(nextNewItemTarget.focusOnNextTarget).toHaveBeenCalled();
+    });
+  });
+
+  describe('#focusOnFirstResultAfterNextSearch', () => {
```

Comment:
This is an async method that uses the `defer` util and returns a promise that resolves to a function that has side effects on the props... this would be tricky to test, and honestly this code smells 🧀

Reply:
I went ahead and implemented the tests instead :D